test/jdk/jdk/crac/LazyProps.java

```diff
@@ -22,34 +22,39 @@
  */
 
 import jdk.crac.*;
-import jdk.test.lib.process.ProcessTools;
-import jdk.test.lib.process.OutputAnalyzer;
+import jdk.test.lib.crac.CracBuilder;
+import jdk.test.lib.crac.CracEngine;
+import jdk.test.lib.crac.CracTest;
 
 /**
  * @test
  * @library /test/lib
  * @run main LazyProps
  */
-public class LazyProps {
-    static public void main(String[] args) throws Exception {
-        if (args.length == 0) {
-            OutputAnalyzer output = ProcessTools.executeTestJvm(
-                    "-XX:CREngine=simengine", "-XX:CRaCCheckpointTo=./cr",
-                    "LazyProps",
-                    "-runTest");
-            output.shouldHaveExitValue(0);
-            output.shouldContain("jdk.crac beforeCheckpoint");
-        } else {
-            Resource resource = new Resource() {
-                @Override
-                public void beforeCheckpoint(Context<? extends Resource> context) throws Exception { }
-                @Override
-                public void afterRestore(Context<? extends Resource> context) throws Exception { }
-            };
-            Core.getGlobalContext().register(resource);
+public class LazyProps implements CracTest {
+    public static void main(String[] args) throws Exception {
+        CracTest.run(LazyProps.class, args);
+    }
```

Comment:
I assume this is the pattern that every test should be following. Is it possible to avoid this boiler plate by e.g. 
```
@run driver CracTest LazyProps
```
and by making CracTest.main to look for the class?

Reply:
Good idea! From docs I didn't really get what's the difference between `@run driver` and `@run main`, do you know?

Reply:
I have not tried that myself, but from jtreg FAQ [1]

> ... `@run driver`. This is the same as `@run main` with the exception that any VM options specified on the command line will not be used when running the specified class.

Since we won't provide VM options, AFAIU there is no effective difference, but using `driver` specifies the intent better.

[1] https://openjdk.org/jtreg/faq.html